src/hotspot/share/cds/archiveUtils.cpp

```diff
@@ -39,6 +39,7 @@
 #include "oops/compressedOops.inline.hpp"
 #include "runtime/arguments.hpp"
 #include "utilities/bitMap.inline.hpp"
+#include "utilities/formatBuffer.hpp"
```

Comment:
Why add the above include?

Reply:
archiveUtils.cpp uses `err_msg`, which is declared in formatBuffer.hpp. Before this fix, this header was included indirectly from heapShared.inline.hpp -> collectedHeap.hpp -> formatBuffer.hpp.

In this PR, I removed the inclusion of collectedHeap.hpp in heapShared.inline.hpp, so the missing dependency was exposed.